src/unstract/sdk/file_storage/impl.py

```diff
@@ -21,10 +21,13 @@ class FileStorage(FileStorageInterface):
     # This class integrates fsspec library for file operations
 
     fs: fsspec  # fsspec file system handle
+    provider: FileStorageProvider
 
     def __init__(self, provider: FileStorageProvider, **storage_config: dict[str, Any]):
         self.fs = FileStorageHelper.file_storage_init(provider, **storage_config)
+        self.provider = provider
 
+    @skip_local_cache
```

Comment:
@gaya3-zipstack Will this annotation, skip everytime read is invoked?

Reply:
The annotation always gets invoked when read() is called.